designcompose/src/main/java/com/android/designcompose/DocServer.kt

```diff
@@ -416,6 +448,7 @@ internal fun DocServer.doc(
 
     // Create a state var to remember the document contents and update it when the doc changes
     val (liveDoc, setLiveDoc) = remember { mutableStateOf<DocContent?>(null) }
+    DesignSettings.designDocStatuses.putIfAbsent(docId, DesignDocStatus())
```

Comment:
Cool didn't know about `putIfAbsent()`. So you are populating an empty entry for `docId` here, and then in `fetchDocuments()` the data is filled out, is that correct? When, if ever, would `DesignSettings.designDocStatuses[id]` be null there?

Reply:
I'm not sure, are you asking about the safe calls below? I'm getting errors if I don't make them safe calls

Reply:
Yes I'm referring to all the safe calls `designDocStatus[docId]?.` but I'm not asking you to remove them. I'm just wondering if you expect them to ever be null and not set anything?